src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +109,12 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        CandidateValidator candidateValidator = new CandidateValidator();
+        return candidates
+                .stream()
+                .filter(candidateValidator::test)
```

Comment:
```suggestion
                .filter(candidateValidator)
```
You can simply pass a predicate implementation to the filter